framework/src/application/node/node.js

```diff
@@ -422,7 +413,10 @@ module.exports = class Node {
 				const transactions = block.transactions
 					.reverse()
 					.map(tx => this.chain.deserializeTransaction(tx));
-				this.transactionPool.onDeletedTransactions(transactions);
+
```

Comment:
instead of looping twice, it's probably good to do below. Also, insertion order doesn't matter anymore so no need to `reverse`
```
for (const transaction of block.transactions) {
  this.transactionPool.add(this.chain.deserializeTransaction(transaction);
}
```